src/CppUTestExt/MockNamedValue.cpp

```diff
@@ -138,6 +144,17 @@ SimpleString MockNamedValue::getType() const
     return type_;
 }
 
+bool MockNamedValue::getBoolValue() const
+{
+    if(type_ == "int")
+        return (value_.intValue_ != 0);
```

Comment:
I covered that in the description.  The particular cases to imagine:

- A C++ test uses `mock().expectOneCall("a").withBoolParameter("x", true).andReturnValue(false);`
The corresponding C implementation uses `return mock_c()->actualCall("a")->withIntParameter("x", true)->intReturnValue();`.

- The reverse case, where C sets an expectation and C++ the actual call.  (This is probably rarer due to the way the library is designed.)

In this variant, the C API cannot use the bool type, so it must use the int type.  The C++ side _could_ use the int type as well but this is weird where the type is logically bool.